diff --git a/helpful.el b/helpful.el
index 20713d2..f2b7c4f 100644
--- a/helpful.el
+++ b/helpful.el
@@ -6,7 +6,7 @@
 ;; URL: https://github.com/Wilfred/helpful
 ;; Keywords: help, lisp
 ;; Version: 0.18
-;; Package-Requires: ((emacs "25") (dash "2.12.0") (dash-functional "1.2.0") (s "1.11.0") (f "0.20.0") (elisp-refs "1.2"))
+;; Package-Requires: ((emacs "25") (dash "2.12.0") (dash-functional "1.2.0") (s "1.11.0") (elisp-refs "1.2"))
 
 ;; This program is free software; you can redistribute it and/or modify
 ;; it under the terms of the GNU General Public License as published by
@@ -48,7 +48,6 @@
 (require 'dash)
 (require 'dash-functional)
 (require 's)
-(require 'f)
 (require 'find-func)
 (require 'nadvice)
 (require 'info-look)
@@ -333,8 +332,8 @@ source code to primitives."
   (let ((emacs-src-dir (read-directory-name "Path to Emacs source code: ")))
     ;; Let the user specify the source path with or without src/,
     ;; which is a subdirectory in the Emacs tree.
-    (unless (equal (f-filename emacs-src-dir) "src")
-      (setq emacs-src-dir (f-join emacs-src-dir "src")))
+    (unless (equal (file-name-nondirectory emacs-src-dir) "src")
+      (setq emacs-src-dir (expand-file-name "src" emacs-src-dir)))
     (setq find-function-C-source-directory emacs-src-dir))
   (helpful-update))
 
@@ -1258,10 +1257,12 @@ LIBRARY-NAME takes the form \"foo.el\" , \"foo.el\" or
 \"src/foo.c\".
 
 If .elc files exist without the corresponding .el, return nil."
-  (when (member (f-ext library-name) '("c" "rs"))
+  (when (member (file-name-extension library-name) '("c" "rs"))
     (setq library-name
-          (f-expand library-name
-                    (f-parent find-function-C-source-directory))))
+          (expand-file-name
+	   library-name
+	   (directory-file-name
+	    (file-name-directory find-function-C-source-directory)))))
   (condition-case nil
       (find-library-name library-name)
     (error nil)))
@@ -2233,7 +2234,7 @@ state of the current symbol."
         ((and source-path primitive-p)
          (format
           "Finding references in a .%s file is not supported."
-          (f-ext source-path)))
+          (file-name-extension source-path)))
         (source-path
          (format "%s is unused in %s."
                  helpful--sym
@@ -2320,7 +2321,7 @@ state of the current symbol."
           (propertize (format "%s Defined in " (if primitive-p "//" ";;"))
                       'face 'font-lock-comment-face)
           (helpful--navigate-button
-           (f-abbrev source-path)
+           (abbreviate-file-name source-path)
            source-path
            pos)
           "\n"))
